hasjob/views/index.py

```diff
@@ -446,7 +442,10 @@ def browse_by_email(md5sum):
     if not md5sum:
         abort(404)
     basequery = JobPost.query.filter_by(md5sum=md5sum)
-    return index(basequery=basequery, md5sum=md5sum, showall=True)
+    if not basequery.count():
```

Comment:
Change to `if basequery.isempty():` since it is a more efficient database call than `.count()`